artifact-manifest/artifact-manifest.md

```diff
@@ -0,0 +1,484 @@
+
+### Deferred Resolution
+
+To support artifact movement to various registry and namespace structures, the registry and path must not be embedded within the artifact definition. Client CLIs and configurations will provide default locations and mappings for where to find the referenced content.
+
+Artifacts that reference other artifacts must include an OCI Artifact Descriptor which includes the `manifest type`, `digest`, `size` and `repo:tag` of the artifact, however it will defer resolution of the reference to client tools that MAY reconstitute the references from multiple repositories and/or registries.
+
+Clients MAY choose to bind to specific digests, assuring they are testing and using the exact graph initially specified. Or, clients may choose to float to a newer version of a tag, benefiting from patches. In all cases, a Notary v2 signature may be used to assure the artifacts are true to their initial ownership and authors.
+
+## Supported Artifact Types
+
+Artifact manifest is intended to support the following artifact types:
+
+### OCI Image
+
+An OCI Image is based on the oci-image-manifest. It's shown as a comparison to the examples below.
+
+![OCI Image](media/wordpress-image-layers.svg)
+
+### Notary v2 Signature Persistance
+
+A Notary v2 signature would persist as a manifest with a config object and a signature, persisted as a blob. However, the signature has no value unto itself. A signature is applied to a given artifact. It's said to be dependent upon another artifact to be complete.
+
+![notary v2 signature](media/notaryv2-signature.svg)
+
+The Notary v2 signature would reference an artifact, such as the `wordpress:v5` image above. Notice the directionality of the references. One or more signatures may be added to a registry after the image was persisted. While an image knows of it's layers, and a Notary v2 signature knows of its config and blob, the Notary v2 signature declares a dependency to the artifact it's signing. The visualization indicates the references through solid lines as these reference types are said to be hard references. Just as the layers of an OCI Image are deleted (*ref-counted -1*), the blobs of a signature are deleted (*ref-counted -1*) when the signature is deleted. Likewise, when an artifact is deleted, the signature would be deleted (*ref-counted -1*) as the signatures have no value without the artifact they are signing.
+
+![wordpress image with layers](media/wordpress-image-layers-sig.svg)
+
+### Helm Charts & CNAB
+
+A Helm chart can represent the images it references within the chart. These references are loose references as they may be persisted in different registries, or may change as a values file is updated. However, the chart may also be persisted together as a collection of artifacts in a registry. The lines are dotted to represent the loose reference. Deleting the `wordpress-chart:v5` may, or may not delete the images as the images have value unto themselves.
```

Comment:
Does it mean the manifest of the chart in this PR only reflects the case when the images and chart are persisted as a collection and the images are NOT loosely referenced?

Reply:
This is an area of churn, and I'm going to create two options for how we can support the two distinct scenarios.
A Notary v2 signature, an SBoM are enhancements to an existing artifact. They `depend-upon` the thing they're enhancing to have meaning. When the artifact they enhance is deleted, these artifacts would also be deleted.

However, in the Helm case, a helm chart can `reference` other images, which _may_, or _may not_ be stored in the same registry. The combination of the digest and `artifact name` are ways to identify it as a unique entity (digest) or stable/updateable tag. The registry will store these values in the manifest, but won't do anything, unless a client tells it to do something.

For instance, compared to the Notary & SBoM example above, a helm chart can be copied to another registry or deleted without impacting the images they reference.
However, the `oci-reg` CLI (imagined) would be able to read the manifest and do a pull and or copy of the helm chart and its referenced images. If the client says `use-digest` a registry can identify the image by its digest as it's unique, regardless of where in the registry it lives. If it says `use-tag`, it would need a [registry.config](https://github.com/notaryproject/nv2/discussions/31) to assist where to find that tag.